minio/signer.py

```diff
@@ -144,7 +147,7 @@ def presign_v4(method, url, credentials,
                                                    new_parsed_url,
                                                    headers_to_sign,
                                                    signed_headers,
-                                                   content_hash_hex)
+                                                   content_sha256_hex)
```

Comment:
This is all there is to add `headers['x-amz-content-sha256']` and also make sure to purge it from the headers after

```
delete(headers, 'x-amz-content-sha256')
```

Reply:
Gotcha! I'll make these changes, likely tomorrow.

Reply:
@harshavardhana one concern that I have is with casing - how do we handle both cases when the user might do:

```python
headers={"X-Amz-Content-Sha256": sha256},
```
versus what you specified:

```python
headers={"x-amz-content-sha256": sha256},
```
We would need to parse through them and make all lower-case every time, which I'm not sure would make sense - the ones that are going into `query` to be signed have variable casing:

```
query["X-Amz-Algorithm"] = _SIGN_V4_ALGORITHM
```

On the other hand, to just expose the variable, we don't need this extra parsing and it just gets added as part of the signature to sign:

```python
    canonical_request = generate_canonical_request(
        method, new_parsed_url, headers_to_sign, signed_headers, content_hash_hex
    )
```
I see where you are coming from in not wanting to add extra variables, but if that is the best practice, then why is the content_hash_hex that currently exists getting special treatment, in terms of being it's own variable to pass into the second function? I think the approach to add it cleanly as a variable (with default as None) would both avoid the capitalization issues or need for extra parsing, and also already match the coding style (having it as a parameter) that already exists. Let me know your thoughts.

Reply:
That is easy every header set here is ultimately canonicalized so we look for lower-cased form and purge 


```py
>>> content_sha256=None
>>> for k in kv.keys():
...     if k.lower() == 'x-amz-content-sha256':
...             content_sha256 = kv.get(k)
...             kv.pop(k)
... 
'help'
>>> print(content_sha256)
help
>>> 
```

Reply:
Where exactly does this happen? So where would you like the check to be? I'm not following because my current usage has the headers passed in as a dictionary (with lower and uppercase) and this quickly gets passed into `get_signed_headers` that does handle the lowercase, but also turns it into a list:

```
def get_signed_headers(headers):
    """
    Get signed headers.

    :param headers: input dictionary to be sorted.
    """
    signed_headers = []
    for header in headers:
        signed_headers.append(header.lower().strip())
    return sorted(signed_headers)
```
so we would need to add an extra loop (as you've shown) to iterate through the dictionary of headers before that (in parse_v4) for this to work. Is this the approach that you would like?

Reply:
You need to add this in the presigned_v4 code @vsoch - if its confusing let me send the PR.